src/vs/platform/workspace/common/workspace.ts

```diff
@@ -140,6 +140,10 @@ export function isSingleFolderWorkspaceIdentifier(obj: unknown): obj is ISingleF
 	return typeof singleFolderIdentifier?.id === 'string' && URI.isUri(singleFolderIdentifier.uri);
 }
 
+export function isEmptyWorkspaceIdentifier(obj: unknown): obj is IEmptyWorkspaceIdentifier {
+	return !isSingleFolderWorkspaceIdentifier(obj) && !isWorkspaceIdentifier(obj);
```

Comment:
This method is dangerous, anything that is not a single-folder-workspace or workspace identifier will match. Rather you should do this:

```ts
export function isEmptyWorkspaceIdentifier(obj: unknown): obj is IEmptyWorkspaceIdentifier {
	if (isSingleFolderWorkspaceIdentifier(obj) || isWorkspaceIdentifier(obj)) {
		return false;
	}

	const emptyWorkspaceIdentifier = obj as IEmptyWorkspaceIdentifier | undefined;

	return typeof emptyWorkspaceIdentifier?.id === 'string';
}
```

Reply:
Yeah. A very good catch. Thanks.